src/components/calcite-combobox/calcite-combobox.tsx

```diff
@@ -242,10 +258,12 @@ export class CalciteCombobox {
   //--------------------------------------------------------------------------
   @State() items: HTMLCalciteComboboxItemElement[] = [];
 
-  @State() selectedItems: HTMLCalciteComboboxItemElement[] = [];
+  @State() selectedItems: (HTMLCalciteComboboxItemElement | SelectedComboboxOption)[] = [];
 
   @State() visibleItems: HTMLCalciteComboboxItemElement[] = [];
 
+  @State() customItems: SelectedComboboxOption[] = [];
```

Comment:
Instead of having another state, should the component just create another ComboBoxElement into the DOM and set it to be selected? Wouldn't that be required to know which items are selected in the component?

Reply:
Maybe. I did it this way because if a dev is using a unidirectional data flow (like react, preact) likely the items are getting constructed based on _their state_. We could add it, but there is a good chance it will just get blown away  on next render. In general I think it's weird for this component to mess with DOM in dev's app. 

Communication of the current selected state should happen via Events. Currently you'll get everything as part of an array in the event detail. The event has a weird name, though `calciteLookupChange`. We could add another event that is more explicit?

Reply:
>  In general I think it's weird for this component to mess with DOM in dev's app.

I agree, but I think there should be a way to get all of the currently selected items without having to listen to events. With the way it is now, if you wanted to know all the selected items, you would have to listen to the events and store that information for later use rather than accessing it when you need it.

For example in a `<select multiple>` element, you can get the selected items through a prop `selectedOptions` or by querying the element for `option`s that are selected.

If we don't want to mess with the DOM within the element, maybe we should just have an event for when a user hits the enter key and there is no match. Then they can create the selected item and insert it themselves.

Reply:
The general "stencil-y" way they recommend data flow is "events up, props down". Using a prop to read child state from a parent feels weird. I could just be brainwashed by React at this point though 😆 

There is also a design question of should that new chip even appear in the list. Not sure maybe @macandcheese has input there

Reply:
We could make an imperative method for getting selected items for this situation, that's another idea.

Reply:
> Using a prop to read child state from a parent feels weird.

I agree with this, which is why I think having the child DOM nodes accessible with their state makes the most sense.

> The general "stencil-y" way they recommend data flow is "events up, props down".

I might be wrong but isn't that more of a framework communication thing? In the end these are all just DOM elements and should function like native ones if possible, right?

Reply:
Regarding chip appearing in the list... I think it’s a nice thing to get “for free”, but I do wonder if that should be an option .. insert-custom-values... 

There may be some validation that needs to occur (a tag may need to be x characters long, for instance) that an app would need to handle before adding the chip and the combo box item to the list (or showing an error message to a user and not adding).

Reply:
ok, I think if we want them to appear in the list, probably the easiest thing to do is construct a new item and just put them in there. I'll update. 

Thanks for feedback @driskull